From: Ryan Tandy <ryan@nardis.ca>
Date: Wed, 1 Jun 2011 09:43:01 -0700
Subject: buffalo-files

---
 files/micro-evtd.event | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/files/micro-evtd.event b/files/micro-evtd.event
index 1510331..93fe5ad 100644
--- a/files/micro-evtd.event
+++ b/files/micro-evtd.event
@@ -113,14 +113,10 @@ button() {
 	    ## Micro IO
 	    case $1 in
 	      1)  stop ;;
-	      8)  /usr/local/bin/initsw.sh ;;
-	      16) /usr/local/bin/ups_shutdown.sh ;;
 	      65) ## Long power button press so force EM
 	          rm -f /boot/hddrootmode
 	          reboot ;;
 	     129) standby 1 ;;
-	     132) /usr/local/bin/initsw.sh ;;
-	     144) /usr/local/bin/ups_shutdown.sh ;;
 	     193) stop ;;
 	     200) micro-evtd -s 013003 ;;
 	     201) micro-evtd -s 013000 ;;
