--- a/files/microapl
+++ b/files/microapl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # microapl Linkstation/Kuro Micro-Controller querry script
 #
@@ -76,15 +76,15 @@
 			  ## Delay amount?
 			  if [ "$2" != "" ] ; then
 			    ## We only do 5 mins intervals
-			    ((COUNT=($2+4)/5))
+			    COUNT=$((($2+4)/5))
 			  fi
 			  LOOP=0
 			  DELAY=0
 			  while [ ${LOOP} -lt ${COUNT} ] ; do
-			    ((DELAY +=5))
-			    ((LOOP ++))
+			    DELAY=$(($DELAY+5))
+			    LOOP=$(($LOOP+1))
 			    /bin/kill -SIGCONT ${PID}
-			    LC=0 ; while [ ${LC} -lt 50 ] ;do ((LC ++)) ; done
+			    LC=0 ; while [ ${LC} -lt 50 ] ;do LC=$((LC+1)) ; done
 			  done
 			  echo "Standby delayed by additional ${DELAY} minutes"
 			else
@@ -397,9 +397,9 @@
 			if [ "$2" == "" ] ; then
 			  echo "#[microapl.hdd_set_power]"
 			  case `$MICRO 803B` in
-			    0) echo -e "hdd12_power=off\nhdd34_power=off" ;;
-			    1) echo -e "hdd12_power=on\nhdd34_power=off" ;;
-			    2) echo -e "hdd12_power=off\nhdd34_power=on" ;;
+			    0) printf "hdd12_power=off\nhdd34_power=off\n" ;;
+			    1) printf "hdd12_power=on\nhdd34_power=off\n" ;;
+			    2) printf "hdd12_power=off\nhdd34_power=on\n" ;;
 			  esac
 			else
 			  TEMP=`printf "0x%02X" $2`
--- a/files/micro-evtd.event
+++ b/files/micro-evtd.event
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # micro-evtd Event Script
 #
 # Sample Event Script written by Bob Perry (2007-2009)
@@ -61,7 +61,7 @@
 dump_info() {
 	## Temp and fan speed info
 	let "rpm = $2 * 30"
-	echo -e "Temp=$1[C-Deg]\nFan=${rpm}rpm\nFan Failures=$3" > $TMP/micro-evtd.status
+	printf "Temp=$1[C-Deg]\nFan=${rpm}rpm\nFan Failures=$3\n" > $TMP/micro-evtd.status
 }
 
 stop() {
@@ -158,7 +158,7 @@
 case "$1" in
 	4)	fan_fault $2 ;;
 	O)	over_heat $2 ;;
-	I)	dump_info $2 $3 ${4//_/ } ;;
+	I)	dump_info $2 $3 $(echo $4 | tr -d _) ;;
 	B)	button $2 $4;;
 	W)	message $2 ;;
 	S)	standby $2 $3 ;;
