--- a/files/micro-evtd.event
+++ b/files/micro-evtd.event
@@ -113,14 +113,10 @@
 	    ## Micro IO
 	    case $1 in
 	      1)  stop ;;
-	      8)  /usr/local/bin/initsw.sh ;;
-	      16) /usr/local/bin/ups_shutdown.sh ;;
 	      65) ## Long power button press so force EM
 	          rm -f /boot/hddrootmode
 	          reboot ;;
 	     129) standby 1 ;;
-	     132) /usr/local/bin/initsw.sh ;;
-	     144) /usr/local/bin/ups_shutdown.sh ;;
 	     193) stop ;;
 	     200) micro-evtd -s 013003 ;;
 	     201) micro-evtd -s 013000 ;;
