--- a/micro_evtd.c
+++ b/micro_evtd.c
@@ -403,8 +403,8 @@
 	char strEventScript[50];
 	
 	// Create the command line
-	sprintf(strEventScript, "/%s/micro_evtd/EventScript %c %d %ld %s %s %d %c", 
-	(CP_SCRIPT ==  cmd? "etc" : strTmpPath), 
+	sprintf(strEventScript, "/%s/micro_evtd.event %c %d %ld %s %s %d %c",
+	(CP_SCRIPT ==  cmd? "usr/sbin" : strTmpPath),
 	cmd, 
 	cmd2, 
 	cmd3,
--- a/Install/EventScript
+++ b/Install/EventScript
@@ -26,8 +26,7 @@
 fi
 
 ## Ensure tmp path is valid
-LEN=${#TMP}-1
-if [ $'\r' = "${TMP:${LEN}:1}" ]; then TMP=${TMP:0:$LEN} ; fi
+TMP=$(echo $TMP | tr -d '\r')
 if [ ! -d ${TMP} ]; then TMP=/tmp ; fi
 
 if [ "$DEBUG" ] && [ -d "$LOG" ]; then
@@ -51,7 +50,7 @@
 create_script() {
 	## Create RAM version of control file
 	if [ ! -d $TMP/micro_evtd ]; then mkdir $TMP/micro_evtd ; fi
-	cp -a /etc/micro_evtd/EventScript $TMP/micro_evtd/.
+	cp -a /usr/sbin/micro_evtd.event $TMP/micro_evtd/.
 	## Stop flashing LEDs and clear error codes
 	micro_evtd -q -s 02520000,02540000
 }
@@ -103,9 +102,13 @@
 	      1)  stop ;;
 	      2)  ;;
 	      4)  ;;
-	      8)  /usr/local/bin/initsw.sh ;;
-	      16) /usr/local/bin/ups_shutdown.sh ;;
+## These files don't exist in a Debian installation
+#	      8)  /usr/local/bin/initsw.sh ;;
+#	      16) /usr/local/bin/ups_shutdown.sh ;;
+	      8)  ;;
+	      16) ;;
 	      65) ## Long power button press
+## Is hddrootmode for EM?
 	          rm -f /boot/hddrootmode
 	          reboot ;;
 	      *)  ;;
@@ -150,7 +153,7 @@
 	4)	fan_fault $2 ;;
 	C)	create_script ;;
 	O)	over_heat $2 ;;
-	I)	TEMP="${4//_/ }"
+	I)	TEMP="$(echo $4 | tr -d _)"
 		dump_info $2 $3 ${TEMP} ;;
 	B)	button $2 $4;;
 	W)	message $2 ;;
